graknlabs/client-nodejs
# Migrate tests to latest rules-nodejs

## What is the goal of this PR?

Our fork of `rules-nodejs` is ~600 commits behind the official upstream repo and it's time to switch to upstream rules so it's easier to keep up.

## What are the changes implemented in this PR?

* Migrate `nodejs_jest_test` to `jasmine_node_test`
  * `it()` instead of `test()` in test definitions
  * `expect(obj).toHaveLength(len)` is not available in Jasmine, so we use `expect(obj.length).toBe(len)`. This corresponds to exact [definition](https://jestjs.io/docs/en/expect.html#tohavelengthnumber) of what original function does.
  * `xit()` instead of `test.skip()`
  * `jasmine.DEFAULT_TIMEOUT_INTERVAL` instead of `jest.setTimeout`
  * `await expectAsync(promise).toBeRejected();` instead of `await expect(promise).rejects.toThrowError();` 
* Use `yarn` to install dependencies instead of `npm`:
  * this is a currently-recommended way by `rules-nodejs` (see their `README`)
  * `grpc` failed to install with `npm_install` — planning to open an issue soon
* Add `node_modules` to `.gitignore` — `rules-nodejs` now install Node deps right into workspace
